fs/mount.c

```diff
@@ -37,15 +37,15 @@ void mount_release(struct mount *mount) {
     unlock(&mounts_lock);
 }
 
-int do_mount(const struct fs_ops *fs, const char *source, const char *point, int flags) {
+int do_mount_with_data(const struct fs_ops *fs, const char *source, const char *point, int flags, void *data) {
```

Comment:
You're supposed to set mount->data after calling this, there shouldn't need to be a new function

Reply:
My problem was (and still is) that do_mount calls `fs->mount` and at that point `startAccessingSecurityScopedResource` must be called. For that, I need the NSURL object.

Reply:
My problem was (and still is) that `do_mount` calls `fs->mount` and at that point `startAccessingSecurityScopedResource` must be called. For that, the `NSURL` object is required.